### PR TITLE
Ip: improve and test numSubnetBitsToSubnetLong

### PR DESCRIPTION
**Pull request by dhalperi**
This function is showing up on profiles. Replace for loop with equivalent
bit-twiddling.

---
This change is [Reviewable](https://reviewable.io/reviews/batfish/batfish/672)

### CODE REVIEW

File: projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/IpTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests of {@link Ip}. */
+@RunWith(JUnit4.class)
+public class IpTest {
+  @Test
+  public void numSubnetBitsToSubnetLong() throws Exception {
```
Comment (dhalperi) [inline projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/IpTest.java]: I tested the first version of this for all numbers 0-31 against the existing implementation.

**Comment by progwriter** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/672#-:-Kz2aQA1AlP1lQBzvOgg:bnfp4nl)*

**Comment by progwriter** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/672)*

**Comment by dhalperi** [comment]
*[projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/IpTest.java, line 14 at r1](https://reviewable.io:443/reviews/batfish/batfish/672#-Kz0eB7XWMGId85rdaEm-r1-14:-Kz2k4lZ8NRVKGjrYzbN:b-896fix) ([raw file](https://github.com/batfish/batfish/blob/8370ec30151e97662542a46ecb7488750153d021/projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/IpTest.java#L14)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

I tested the first version of this for all numbers 0-31 against the existing implementation.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/672)*

**Pull request merged**